sdk/textanalytics/Azure.AI.TextAnalytics/tests/RecognizeHealthcareEntitiesTests.cs

```diff
@@ -261,6 +261,41 @@ public async Task RecognizeHealthcareEntitiesBatchWithStatisticsTest()
             Assert.AreEqual(0, resultCollection.Statistics.InvalidDocumentCount);
         }
 
+        [Test]
+        public async Task RecognizeHealthcareEntitiesBatchWithCancellationAsync()
+        {
+            TextAnalyticsClient client = GetClient();
+            string document = @"RECORD #333582770390100 | MH | 85986313 | | 054351 | 2/14/2001 12:00:00 AM | CORONARY ARTERY DISEASE | Signed | DIS | \
+                                Admission Date: 5/22/2001 Report Status: Signed Discharge Date: 4/24/2001 ADMISSION DIAGNOSIS: CORONARY ARTERY DISEASE. \
+                                HISTORY OF PRESENT ILLNESS: The patient is a 54-year-old gentleman with a history of progressive angina over the past several months. \
+                                The patient had a cardiac catheterization in July of this year revealing total occlusion of the RCA and 50% left main disease ,\
+                                with a strong family history of coronary artery disease with a brother dying at the age of 52 from a myocardial infarction and \
+                                another brother who is status post coronary artery bypass grafting. The patient had a stress echocardiogram done on July , 2001 , \
+                                which showed no wall motion abnormalities , but this was a difficult study due to body habitus. The patient went for six minutes with \
+                                minimal ST depressions in the anterior lateral leads , thought due to fatigue and wrist pain , his anginal equivalent. Due to the patient's \
+                                increased symptoms and family history and history left main disease with total occasional of his RCA was referred for revascularization with open heart surgery.";
+
+            var batchDocuments = new List<string>();
+
+            for (var i = 0; i < 10; i++)
+            {
+                batchDocuments.Add(document);
+            }
+
+            HealthcareOperation operation = await client.StartHealthcareBatchAsync(batchDocuments, "en");
+
+            await operation.CancelAsync();
+
+            RequestFailedException ex = Assert.ThrowsAsync<RequestFailedException>(async() => await operation.WaitForCompletionAsync());
+            Assert.AreEqual("The operation was canceled so no value is available.", ex.Message);
+
+            Assert.IsTrue(operation.HasCompleted);
+            Assert.IsFalse(operation.HasValue);
```

Comment:
it will be good to also add one for `operation.Value` to make sure we throw the same exception than the one from `await operation.WaitForCompletionAsync()`.